lib/Dialect/Sim/SimDialect.cpp

```diff
@@ -40,6 +41,12 @@ Operation *SimDialect::materializeConstant(::mlir::OpBuilder &builder,
                                            ::mlir::Type type,
                                            ::mlir::Location loc) {
 
+  // Delegate non 'sim' types to the HW dialect materializer.
```

Comment:
I'm curious why this is necessary?  Are hw constant requests really winding up here?  Why?